roles/nginx-single-host/tasks/main.yml

```diff
@@ -0,0 +1,84 @@
+---
+# This role sets up nginx, and the proper config, for
+# a planetary room instance where all the services are on a
+# single host and share a domain. For example: the host planet.fun
+# has go-ssb-room, planetary-graphql, and rooms-frontend all running
+# on it and they are accessible via planet.name/login, planet.name, and
+# planet.name/graphql respectively
+
+- name: Update and upgrade apt packages
+  apt:
+    upgrade: yes
+    update_cache: yes
+    cache_valid_time: 86400
+
+
+- name: Install nginx and certbot
+  apt:
+    pkg:
+      - nginx
+      - certbot
+      - python3-certbot-nginx
+    state: latest
+
+
+- name: Confirm nginx conf folder exists
+  ansible.builtin.stat:
+    path: "/etc/nginx/conf.d"
+  register: conf_d
+
+# This sets the connection_upgrade var used in our site config
+- name: Copy websocket_upgrade.conf to conf.d
+  ansible.builtin.copy:
+    src: websocket_upgrade.conf
+    dest: "/etc/nginx/conf.d/websocket_upgrade.conf"
+  when: conf_d.stat.exists
+
+
+- name: Build and copy over site config
+  ansible.builtin.template:
+    src: site.conf.tpl
+    dest: "/etc/nginx/sites-available/{{ inventory_hostname }}"
+
+
+- name: remove default site config
+  file:
+    path: /etc/nginx/sites-enabled/default
+    state: absent
+
+
+- name: symlink site config to sites-enabled
+  file:
+    src: "/etc/nginx/sites-available/{{ inventory_hostname }}"
+    dest: "/etc/nginx/sites-enabled/{{ inventory_hostname }}"
+    state: link
+
+
+- name: Ensure config is valid
+  ansible.builtin.command: nginx -t
+  changed_when: false
+  notify: restart nginx
+
+
+- name: "Print out certbot instructions"
+  ansible.builtin.debug:
+    msg: "Successfully set up the host!  There are some manual steps to be done on the server. SSH in and read /root/success.txt"
+
+- name: success to file
+  ansible.builtin.copy:
+    dest: /root/success.txt
+    content: |
+      Congratulations! You've got everything set up and are almost done.
+      The only thing left is to run certbot so your site is actually up.
+      This is best done manually on the host server.
```

Comment:
Ah, i should say it's best done manually "right now".  Two of the commands being run are interactive, and require not only answering some questions but going to your dns provider and adding new records based on output from the commands.  

we have a ticket(https://github.com/planetary-social/ansible-scripts/issues/16) for working with the API of our DNS so that this step _can_ be done as an ansible script.  When that work is finished, we can remove the step here.